lib/adagrams.rb

```diff
@@ -0,0 +1,218 @@
+# THIS PROGRAM WE WROTE PASSES 16/16 OF THE TESTS BUT DOES NOT FUNCTION PROPERLY IN THE TERMINAL. IF YOU UNCOMMENT LINE 61, THIS GAME RUNS PROPERLY IN THE TERMINAL BUT ONLY PASSES 15/16 TESTS.
+
+@words_created = []
+@final_winner = []
+
+@all_points = []
+in_progress = true
+
+
+while in_progress
+  in_progress = false
+
+  def draw_letters()
+    letters =
+    ["a", "a", "a", "a", "a", "a", "a", "a", "a", "b", "b", "c", "c", "d", "d", "d", "d", "e", "e", "e", "e", "e", "e", "e", "e", "e", "e", "e", "e", "f", "f", "g", "g", "g", "h", "h", "i", "i", "i", "i", "i", "i", "i", "i", "i", "j", "k", "l", "l", "l", "l", "m", "m", "n", "n", "n", "n", "n", "n", "o", "o", "o", "o", "o", "o", "o", "o", "p", "p", "q", "r", "r", "r", "r", "r", "r", "s", "s", "s", "s", "t", "t", "t", "t", "t", "t", "u", "u", "u", "u", "v", "v", "x", "y", "y", "z"]
+
+    pool = letters.map do |element|
+      element.upcase
+    end
+
+    pool = pool.shuffle
+
+    hand_drawn = []
+
+    10.times do |i|
+      hand_drawn << pool[i]
+    end
+
+    return hand_drawn
+  end
+
+
+  def uses_available_letters?(input, letters_drawn_input)
+    words_created = []
+
+    input_array = []
+    input.length.times do |i|
+      input_array << input[i]
+    end
+
+    testing_array = []
+    editted_letters_drawn = letters_drawn_input
+
+    input_array.length.times do |i|
+      if editted_letters_drawn.include?(input_array[i])
+        index_of_letter = editted_letters_drawn.index(input_array[i])
+        editted_letters_drawn.delete_at(index_of_letter)
+
+        testing_array << true
+
+      else
+
+        testing_array << false
+      end
+
+    end
+
+
+
+    if testing_array.all?
+      # @words_created << input
+      return true
+    else
+      return false
+    end
```

Comment:
In this method, you all create `testing_array` to represent an array of booleans (`true`s or `false`s) for each letter if the letter was found... if the letter was not found, you put into `testing_array` a `false` value.

In the end, you ask `if testing_array.all?`, which asks, "if the testing array is all true values, this is true".

We know that using `return` will stop a method and return a value. Specifically, we know that if a code reaches a line and executes `return false`, it will stop proceeding into the method and get out of the method and return `false`.


What if instead of approaching this problem with "Let's get an array of trues and falses, and if there's a false, then the answer "uses_available_letters?" is false"... We approach it with, "if we need to shovel in `false` into `testing_array` once, then we know that the answer is `false`"?

For example, on line 53, you all wrote `testing_array << false`. Is there any case in which shoveling `false` into `testing_array` does _not_ mean that the method should return false overall?